contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditLanguageModal.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+
+  <KModal
+    :title="$tr('editLanguage')"
+    :submitText="$tr('saveAction')"
+    :cancelText="$tr('cancelAction')"
+    data-test="edit-language-modal"
+    :submitDisabled="!selectedLanguage"
+    @submit="handleSave"
+    @cancel="close"
+  >
+    <p data-test="resources-selected-message">
+      {{ $tr('resourcesSelected', { count: nodeIds.length }) }}
+    </p>
+    <p v-if="isMultipleNodeLanguages" data-test="different-languages-message">
+      {{ $tr('differentLanguages') }}
+    </p>
+    <KTextbox
+      v-model="searchQuery"
+      autofocus
+      data-test="search-input"
+      :label="$tr('selectLanguage')"
+      style="margin-top: 0.5em"
+    />
+    <template v-if="isTopicSelected">
+      <KCheckbox
+        v-model="updateDescendants"
+        data-test="update-descendants-checkbox"
+        :label="$tr('updateDescendantsCheckbox')"
+      />
+      <hr
+        :style="dividerStyle"
+      >
+    </template>
+    <div
+      ref="languages"
+      class="languages-options"
+      data-test="language-options-list"
+    >
+      <KRadioButton
+        v-for="language in languageOptions"
+        :key="language.id"
+        v-model="selectedLanguage"
+        :value="language.id"
+        :label="languageText(language)"
+      />
+    </div>
+  </KModal>
+
+</template>
+
+
+<script>
+
+  import { mapGetters, mapActions } from 'vuex';
+  import { LanguagesList } from 'shared/leUtils/Languages';
+  import { ContentKindsNames } from 'shared/leUtils/ContentKinds';
+
+  export default {
+    name: 'EditLanguageModal',
+    props: {
+      nodeIds: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        selectedLanguage: '',
+        searchQuery: '',
+        updateDescendants: false,
+      };
+    },
+    computed: {
+      ...mapGetters('contentNode', ['getContentNodes']),
+      nodes() {
+        return this.getContentNodes(this.nodeIds);
+      },
+      isTopicSelected() {
+        return this.nodes.some(node => node.kind === ContentKindsNames.TOPIC);
+      },
+      isMultipleNodeLanguages() {
+        const languages = new Set(this.nodes.map(node => node.language).filter(Boolean));
+        return languages.size > 1;
+      },
+      languageOptions() {
+        const searchQuery = this.searchQuery.trim().toLowerCase();
+        if (!this.searchQuery) {
+          return LanguagesList;
+        }
+        const criteria = ['id', 'native_name', 'readable_name'];
+        return LanguagesList.filter(lang =>
+          criteria.some(key => lang[key]?.toLowerCase().includes(searchQuery))
```

Comment:
It's worth-noting that the chaining operator(`?.`) is [not supported in IE 11](https://caniuse.com/?search=%3F.) that we are still supporting(The last time i checked) :(.